liblumen_alloc gc test fail to compile

In https://github.com/lumen/lumen/runs/992513556

```
ompiling liblumen_alloc v0.1.0 (/__w/lumen/lumen/liblumen_alloc)
2020-08-17T07:28:09.8367431Z error[E0277]: the trait bound `erts::process::gc::rootset::RootSet: std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm; 3]>` is not satisfied
2020-08-17T07:28:09.8368477Z    --> liblumen_alloc/src/erts/process/gc/tests/collector.rs:149:13
2020-08-17T07:28:09.8368672Z     |
2020-08-17T07:28:09.8368809Z 149 |     process.garbage_collect(0, &mut roots).unwrap();
2020-08-17T07:28:09.8369017Z     |             ^^^^^^^^^^^^^^^ the trait `std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm; 3]>` is not implemented for `erts::process::gc::rootset::RootSet`
2020-08-17T07:28:09.8369181Z     |
2020-08-17T07:28:09.8369312Z     = help: the following implementations were found:
2020-08-17T07:28:09.8369484Z               <erts::process::gc::rootset::RootSet as std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm]>>
2020-08-17T07:28:09.8369682Z               <erts::process::gc::rootset::RootSet as std::convert::From<std::vec::Vec<erts::term::boxed::Boxed<erts::term::arch::arch_x86_64::RawTerm>>>>
2020-08-17T07:28:09.8369894Z     = note: required because of the requirements on the impl of `std::convert::Into<erts::process::gc::rootset::RootSet>` for `&mut [erts::term::arch::arch_x86_64::RawTerm; 3]`
2020-08-17T07:28:09.8370001Z 
2020-08-17T07:28:09.8395941Z error[E0277]: the trait bound `erts::process::gc::rootset::RootSet: std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm; 3]>` is not satisfied
2020-08-17T07:28:09.8396629Z    --> liblumen_alloc/src/erts/process/gc/tests/collector.rs:156:13
2020-08-17T07:28:09.8396770Z     |
2020-08-17T07:28:09.8396902Z 156 |     process.garbage_collect(0, &mut roots).unwrap();
2020-08-17T07:28:09.8397088Z     |             ^^^^^^^^^^^^^^^ the trait `std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm; 3]>` is not implemented for `erts::process::gc::rootset::RootSet`
2020-08-17T07:28:09.8397243Z     |
2020-08-17T07:28:09.8397369Z     = help: the following implementations were found:
2020-08-17T07:28:09.8397729Z               <erts::process::gc::rootset::RootSet as std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm]>>
2020-08-17T07:28:09.8397920Z               <erts::process::gc::rootset::RootSet as std::convert::From<std::vec::Vec<erts::term::boxed::Boxed<erts::term::arch::arch_x86_64::RawTerm>>>>
2020-08-17T07:28:09.8398116Z     = note: required because of the requirements on the impl of `std::convert::Into<erts::process::gc::rootset::RootSet>` for `&mut [erts::term::arch::arch_x86_64::RawTerm; 3]`
2020-08-17T07:28:09.8398222Z 
2020-08-17T07:28:09.8707958Z error[E0277]: the trait bound `erts::process::gc::rootset::RootSet: std::convert::From<&mut [_; 0]>` is not satisfied
2020-08-17T07:28:09.8708587Z    --> liblumen_alloc/src/erts/process/gc/tests/collector.rs:300:13
2020-08-17T07:28:09.8708723Z     |
2020-08-17T07:28:09.8708854Z 300 |     process.garbage_collect(0, &mut roots).unwrap();
2020-08-17T07:28:09.8709019Z     |             ^^^^^^^^^^^^^^^ the trait `std::convert::From<&mut [_; 0]>` is not implemented for `erts::process::gc::rootset::RootSet`
2020-08-17T07:28:09.8709180Z     |
2020-08-17T07:28:09.8709304Z     = help: the following implementations were found:
2020-08-17T07:28:09.8709673Z               <erts::process::gc::rootset::RootSet as std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm]>>
2020-08-17T07:28:09.8709867Z               <erts::process::gc::rootset::RootSet as std::convert::From<std::vec::Vec<erts::term::boxed::Boxed<erts::term::arch::arch_x86_64::RawTerm>>>>
2020-08-17T07:28:09.8710059Z     = note: required because of the requirements on the impl of `std::convert::Into<erts::process::gc::rootset::RootSet>` for `&mut [_; 0]`
2020-08-17T07:28:09.8710165Z 
2020-08-17T07:28:09.8887400Z error[E0277]: the trait bound `erts::process::gc::rootset::RootSet: std::convert::From<&mut [_; 0]>` is not satisfied
2020-08-17T07:28:09.8888022Z    --> liblumen_alloc/src/erts/process/gc/tests/collector.rs:389:13
2020-08-17T07:28:09.8888168Z     |
2020-08-17T07:28:09.8888304Z 389 |     process.garbage_collect(0, &mut roots).unwrap();
2020-08-17T07:28:09.8888499Z     |             ^^^^^^^^^^^^^^^ the trait `std::convert::From<&mut [_; 0]>` is not implemented for `erts::process::gc::rootset::RootSet`
2020-08-17T07:28:09.8888651Z     |
2020-08-17T07:28:09.8888794Z     = help: the following implementations were found:
2020-08-17T07:28:09.8888965Z               <erts::process::gc::rootset::RootSet as std::convert::From<&mut [erts::term::arch::arch_x86_64::RawTerm]>>
2020-08-17T07:28:09.8889153Z               <erts::process::gc::rootset::RootSet as std::convert::From<std::vec::Vec<erts::term::boxed::Boxed<erts::term::arch::arch_x86_64::RawTerm>>>>
2020-08-17T07:28:09.8889349Z     = note: required because of the requirements on the impl of `std::convert::Into<erts::process::gc::rootset::RootSet>` for `&mut [_; 0]`
2020-08-17T07:28:09.8889447Z 
2020-08-17T07:28:11.2576613Z error: aborting due to 4 previous errors
```

--- a/liblumen_alloc/src/erts/process/gc/tests/collector.rs
+++ b/liblumen_alloc/src/erts/process/gc/tests/collector.rs
@@ -5,6 +5,7 @@ use core::ops::Deref;
 use core::ptr::NonNull;
 
 use crate::erts::process::alloc::TermAlloc;
+use crate::erts::process::gc::RootSet;
 use crate::erts::process::test::process;
 use crate::erts::term::closure::*;
 use crate::erts::term::prelude::*;
@@ -146,14 +147,16 @@ fn simple_gc_test(process: Process) {
 
     // Run garbage collection
     let mut roots = [tuple_term, list_term, closure_term];
-    process.garbage_collect(0, &mut roots).unwrap();
+    let root_set = RootSet::new(&mut roots);
+    process.garbage_collect(0, root_set).unwrap();
     process.set_flags(ProcessFlags::NeedFullSweep);
     // Grab post-collection size
     let collected_size_first = process.young_heap_used();
 
     //dbg!(process.acquire_heap().heap().young_generation());
     // Run it again and make sure the heap size stays the same
-    process.garbage_collect(0, &mut roots).unwrap();
+    let root_set = RootSet::new(&mut roots);
+    process.garbage_collect(0, root_set).unwrap();
     let collected_size_second = process.young_heap_used();
     assert_eq!(collected_size_first, collected_size_second);
 
@@ -297,7 +300,8 @@ fn tenuring_gc_test(process: Process, _perform_fullsweep: bool) {
     let peak_size = process.young_heap_used();
     // Run first garbage collection
     let mut roots = [];
-    process.garbage_collect(0, &mut roots).unwrap();
+    let root_set = RootSet::new(&mut roots);
+    process.garbage_collect(0, root_set).unwrap();
 
     // Verify size of garbage collected meets expectation
     let collected_size = process.young_heap_used();
@@ -386,7 +390,8 @@ fn tenuring_gc_test(process: Process, _perform_fullsweep: bool) {
     // allocated
     let second_peak_size = process.young_heap_used();
     let mut roots = [];
-    process.garbage_collect(0, &mut roots).unwrap();
+    let root_set = RootSet::new(&mut roots);
+    process.garbage_collect(0, root_set).unwrap();
 
     // Verify no garbage was collected, we should have just tenured some data,
     // the only data on the young heap should be a single cons cell

